src/scripts/mongo/change_password.js

```diff
@@ -15,10 +15,10 @@ function changePassword(email, newPassword, callback) {
         return callback(err);
       }
 
-      users.update({ email: email }, { $set: { password: hash } }, function (err, count) {
+      users.update({ email: email }, { $set: { password: hash } }, function (err, writeResult) {
         client.close();
         if (err) return callback(err);
-        callback(null, count > 0);
+        callback(null, writeResult.result.nModified > 0);
```

Comment:
Doing the whole `writeResult.result.nModified` chain call seems a bit dangerous given that we don't know if there is a condition where mongo might not return some of those objects producing a TypeError. Maybe we can do this in a safer way, something like:
```js
const modified = writeResult && writeResult.result && writeResult.result.nModified;
callback(null, writeResult.result.nModified > 0);
```

Reply:
what should happen if modified is null | undefined?